### ISSUE TITLE
Clean-up dependencies in environment.yaml

### ISSUE THREAD

**Issue by adverley** [Victorlouisdg/linen#1]
Hard to imagine linen uses pytorch, cudatoolkit etc.

**Comment by Victorlouisdg** [comment]
True, those are from the [research-template](https://github.com/tlpss/research-template). I guess the real question is "what is our purpose of the `environment.yaml`"?

**Comment by adverley** [comment]
Speedup setup process for fellow devs who want to contribute to `linen`.
